charts/dns/templates/external-dns-deployment.yaml

```diff
@@ -5,45 +5,112 @@ metadata:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.priorityClassName }}
+      priorityClassName: {{ . | quote }}
+      {{- end }}
+      {{- with .Values.dnsPolicy }}
+      dnsPolicy: {{ . }}
+      {{- end }}
       containers:
         - name: external-dns
-          image: "{{ .Values.externalDNS.repository }}:{{ .Values.externalDNS.tag }}"
+          image: "{{ .Values.externalDNS.registry }}/{{ .Values.externalDNS.repository }}:{{ .Values.externalDNS.tag }}"
           imagePullPolicy: {{ .Values.externalDNS.pullPolicy }}
-          args:
-          - --source=ingress
-          - --source=service
-        {{- if default false .Values.externalDNS.sourceIstioObjects }}
-          - --source=istio-gateway
-          - --source=istio-virtualservice
-        {{- end }}
-          {{- range $zonename, $zonefile := .Values.pdns.zones }}
-          - --domain-filter={{ $zonename }}
-          {{- end }}
-          envFrom:
-          - configMapRef:
-              name: {{ include "externalDNS.fullname" . }}
+          {{- with .Values.externalDNS.securityContext }}
           securityContext:
-            runAsNonRoot: true
-            runAsUser: 65534
-            readOnlyRootFilesystem: true
-            capabilities:
-              drop: ["ALL"]
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          args:
+            - --log-level={{ .Values.externalDNS.logLevel }}
+            - --log-format={{ .Values.externalDNS.logFormat }}
+
+            - --interval={{ .Values.externalDNS.interval }}
+            {{- if .Values.externalDNS.triggerLoopOnEvent }}
+            - --events
+            {{- end }}
+
+            {{- range .Values.externalDNS.sources }}
+            - --source={{ . }}
+            {{- end }}
+            {{- range $zonename, $zonefile := .Values.zones }}
+            - --domain-filter={{ $zonename }}
+            {{- end }}
+            - --managed-record-types={{ .Values.externalDNS.managedRecordTypes }}
+ 
+            - --registry=txt
+            {{- if .Values.externalDNS.txtOwnerId }}
+            - --txt-owner-id={{ .Values.externalDNS.txtOwnerId }}
+            {{- end }}
+            {{- if .Values.externalDNS.txtPrefix }}
+            - --txt-prefix={{ .Values.externalDNS.txtPrefix }}
+            {{- end }}
+            {{- if .Values.externalDNS.txtSuffix }}
+            - --txt-suffix={{ .Values.externalDNS.txtSuffix }}
+            {{- end }}
+            {{- if .Values.externalDNS.txtWildcardReplacement }}
+            - --txt-wildcard-replacement={{ .Values.externalDNS.txtWildcardReplacement }}
+            {{- end }}
+
+            - --provider=pdns
+            - --pdns-server=http://{{ include "pdns.fullname" . }}-http
+            - --pdns-api-key=$(POWERDNS_API_KEY)
+
+          {{/*
+            - --pdns-tls-enabled=true/false
+            - --tls-ca < required
+            - --tls-client-cert < optional
+            - --tls-client-cert-key < optional
```

Comment:
These flags seem odd. Copy/paste artifact perhaps?

Reply:
These flag exist, but they are in a Helm comment. I expected you to read them and let me know what you think about them. Should I publish them somehow?

We are not using them at the moment as we do the TLS endpoint at ingress level.

Reply:
Wow, my brain completely missed the comment line there :woopsie:

I don't think any sane person would want to do tls termination on the application when running on k8s. It's easy to add it if someone asks for it.